src/ServiceDefinition.php

```diff
@@ -23,6 +24,11 @@
     private $singleton;
 
     /**
+     * @var bool
+     */
+    private $factory;
```

Comment:
@tomphp 

Consistent with what we got, but should we consider prepending `is` to these boolean fields?

Wasn't quite sure at first whether this was supposed to be the name of the factory.